src/messages/text_document_hover.cc

```diff
@@ -3,33 +3,34 @@
 
 namespace {
 
+static std::string getHoverString(const optional<std::string> &hover,
+                                  const std::string &comments,
+                                  const std::string &detailed_name) {
+  // TODO: Properly format multi-line comments.
+  std::string hover_string = comments;
+  if (hover)
```

Comment:
The newline may be undesired in Emacs lsp-mode and LanguageClient-neovim .

Reply:
Yeah, I'm planning to return two strings as a response for the hover request. One for the formatted comment and one for the actual description/signature.

Both returned strings are lsMarkedString strings, i.e. markup strings. The comment string should have the markup type "text". The description/signature string should have the markup type "c++" to be properly rendered.